spec/models/order_spec.rb

```diff
@@ -29,5 +29,47 @@
       order.save!
       expect(order.postage).to eq 600
     end
+
+    describe 'delivery_fee' do
+      example '商品の合計が0円から10000円未満だと300円の代引き手数料がかかる' do
+        product.update!(price_without_tax: (9999 / (110.0 / 100.0)).ceil)
+        create :cart_item, cart:, product:, quantity: 1
+        order.save!
+        expect(order.delivery_fee).to eq 300
+      end
+
+      example '商品の合計が10000円から30000円未満だと400円の代引き手数料がかかる' do
+        product.update!(price_without_tax: (10000 / (110.0 / 100.0)).ceil)
+        create :cart_item, cart:, product:, quantity: 1
+        order.save!
+        expect(order.delivery_fee).to eq 400
+
+        product.update!(price_without_tax: (29999 / (110.0 / 100.0)).ceil)
+        cart = user.create_cart!
+        create :cart_item, cart:, product:, quantity: 1
+        order.save!
+        expect(order.delivery_fee).to eq 400
```

Comment:
factoryをうまくつかってコードを減らして、何を検証しているかを簡潔にしたい。
```suggestion
        order1 = create(:order, order_items: [create(:order_item, price_without_tax: 10_000)])
        expect(order1.delivery_fee).to eq 400

        order2 = create(:order, order_items: [create(:order_item, price_without_tax: 29_999)])
        expect(order2.delivery_fee).to eq 400
```

もしくは、order_itemのfactoryに association :order を追加するともう少しコンパクトになるかも（動作確認はしてないです）

```rb
        order_item1 = create(order_item: price_without_tax: 10_000)
        expect(order_item1.order.delivery_fee).to eq 400

        order_item2 = create(order_item: price_without_tax: 29_999)
        expect(order_item2.order.delivery_fee).to eq 400
```

Reply:
全体的にDRYにしておきたい。